tinygrad/ops.py

```diff
@@ -183,16 +182,29 @@ def backward(ctx, grad_output):
     rcout = cout//ctx.groups
 
     dx, dw = np.zeros_like(x), np.zeros_like(w)
-    for g in range(ctx.groups):
-      tw = w[g*rcout:(g*rcout+rcout)].reshape(rcout, -1)
-      for Y in range(grad_output.shape[2]):
-        for X in range(grad_output.shape[3]):
+
+    gdx = dx.reshape(bs,ctx.groups,cin,x.shape[2],x.shape[3])
+    gdw = dw.reshape(ctx.groups,rcout, cin, H, W)
+
+    ggg = grad_output.reshape(grad_output.shape[0],ctx.groups,rcout,oy,ox)
+
+    gx = x.reshape(x.shape[0],ctx.groups,cin,x.shape[2],x.shape[3])
+    str_x = np.lib.stride_tricks.as_strided(
+                gx,
+                shape=(bs, ctx.groups, cin, oy, ox, H, W),
+                strides=(gx.strides[0], gx.strides[1], gx.strides[2], gx.strides[3]*ys, gx.strides[4]*xs, gx.strides[3], gx.strides[4]),
+                writeable=False, 
+            )
+
+    gdw = np.einsum('igkYX,igjYXyx -> gkjyx',ggg,str_x)
+   
+    #needs to be optimized
```

Comment:
Yea, I had trouble with this too when I wrote the fast convs. It's the stupid add accumulation I don't know how to do fast.

Reply:
probably can be done with strides, have to think, but you can not just einsum into a strided array (even if you could you shouldn't)